src/remote/remote.go

```diff
@@ -359,15 +359,9 @@ func (c *Client) Retrieve(target *core.BuildTarget) (*core.BuildMetadata, error)
 			tree := &pb.Tree{}
 			if err := c.readByteStreamToProto(dir.TreeDigest, tree); err != nil {
 				return err
-			}
-			if err := c.downloadDirectory(dirPath, tree.Root); err != nil {
+			} else if err := c.downloadDirectory(dirPath, tree.Root); err != nil {
 				return err
 			}
```

Comment:
Not sure I understand how the `if...else` here is different from before.

Reply:
oh it's not, this is after some other changes were made and then unmade.

The actual fix is deleting lines 366-370.